internal/web/server.go

```diff
@@ -232,351 +262,416 @@ func (s *server) Start(ctx context.Context) error {
+		if pkgName != "" {
+			var pruned []string
+			var err error
+			// dependency checks are only necessary for clusterpackages, as there are no dependencies on namespaced packages
+			if g, err1 := s.dependencyMgr.NewGraph(r.Context()); err1 != nil {
+				err = fmt.Errorf("error validating uninstall: %w", err1)
+			} else {
+				g.Delete(pkgName)
+				pruned = g.Prune()
+				if err1 := g.Validate(); err1 != nil {
+					err = fmt.Errorf("%v cannot be uninstalled: %w", pkgName, err1)
+				}
+			}
+			err = s.templates.pkgUninstallModalTmpl.Execute(w, map[string]any{
+				"PackageName": pkgName,
+				"Pruned":      pruned,
+				"Err":         err,
+				"PackageHref": util.GetClusterPkgHref(pkgName),
+			})
+			checkTmplError(err, "pkgUninstallModalTmpl")
+		} else {
+			err := s.templates.pkgUninstallModalTmpl.Execute(w, map[string]any{
+				"Namespace":   namespace,
+				"Name":        name,
+				"PackageHref": util.GetNamespacedPkgHref(manifestName, namespace, name),
+			})
+			checkTmplError(err, "pkgUninstallModalTmpl")
+		}
 	}
 }
 
-func (s *server) uninstallModal(w http.ResponseWriter, r *http.Request) {
-	pkgName := r.FormValue("packageName")
-	var pruned []string
-	var err error
-	if g, err1 := s.dependencyMgr.NewGraph(r.Context()); err1 != nil {
-		err = fmt.Errorf("error validating uninstall: %w", err1)
+func (s *server) open(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	pkgName := mux.Vars(r)["pkgName"]
+	namespace := mux.Vars(r)["namespace"]
+	name := mux.Vars(r)["name"]
+
+	if pkgName != "" {
+		var pkg v1alpha1.ClusterPackage
+		if err := s.pkgClient.ClusterPackages().Get(ctx, pkgName, &pkg); err != nil {
+			s.respondAlertAndLog(w, err, "Could not get ClusterPackage", "danger")
+			return
+		}
+		s.handleOpen(ctx, w, &pkg)
 	} else {
-		g.Delete(pkgName)
-		pruned = g.Prune()
-		if err1 := g.Validate(); err1 != nil {
-			err = fmt.Errorf("%v cannot be uninstalled: %w", pkgName, err1)
+		var pkg v1alpha1.Package
+		if err := s.pkgClient.Packages(namespace).Get(ctx, name, &pkg); err != nil {
+			s.respondAlertAndLog(w, err, "Could not get Package", "danger")
+			return
 		}
+		s.handleOpen(ctx, w, &pkg)
 	}
-	err = s.templates.pkgUninstallModalTmpl.Execute(w, map[string]any{
-		"PackageName": pkgName,
-		"Pruned":      pruned,
-		"Err":         err,
-	})
-	checkTmplError(err, "pkgUninstallModalTmpl")
 }
 
-func (s *server) uninstall(w http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-	pkgName := r.FormValue("packageName")
-	var pkg v1alpha1.ClusterPackage
-	if err := s.pkgClient.ClusterPackages().Get(ctx, pkgName, &pkg); err != nil {
-		s.respondAlertAndLog(w, err, fmt.Sprintf("An error occurred fetching %v during uninstall", pkgName), "danger")
-		return
+func (s *server) handleOpen(ctx context.Context, w http.ResponseWriter, pkg ctrlpkg.Package) {
+	fwName := pkg.GetName()
+	if pkg.IsNamespaceScoped() {
+		fwName = cache.ObjectName{Namespace: pkg.GetNamespace(), Name: pkg.GetName()}.String()
```

Comment:
This can be simplified, because the `ObjectName` does not put a `/` if the namespace is empty, resulting in the same outcome.